podcasts/Login/GoogleSocialLogin.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import PocketCastsServer
+import GoogleSignIn
+
+enum GoogleSocialLoginError: Error {
+    case emptyIdToken
+}
+
+class GoogleSocialLogin {
```

Comment:
This is super specific for now.

My plan when adding Apple is that we'll have a protocol and all login providers will implement the necessary methods.